easyfix-messages/easyfix-messages-gen/src/gen.rs

```diff
@@ -498,6 +481,7 @@ impl Generator {
                     } else {
                         return Err(DeserializeError::GarbledMessage("MsgType<35> not third tag".into()));
                     };
+                    deserializer.set_msg_type(msg_type);
```

Comment:
`deserializer.deserialize_msg_type()` sets msg_type, so there is no need to call this again. That's why `set_msg_type` had `debug_assert` to prevent doing the same twice.